### PR TITLE
feat: support force flag for osctl kubeconfig

### PR DESCRIPTION
**Pull request by rsmitty**
This PR allows you to specify the kubeconfig command like `osctl
kubeconfig . --force` or `osctl kubeconfig . -f` to delete the
kubeconfig file before we write it out.

Signed-off-by: Spencer Smith <robertspencersmith@gmail.com>

### CODE REVIEW

File: cmd/osctl/cmd/kubeconfig.go [hunk 1]
```diff
@@ -56,11 +60,19 @@ Kubeconfig will be written to PWD/kubeconfig or [local-path]/kubeconfig if speci
 				localPath = args[0]
 			}
 
+			//Drop the existing kubeconfig before writing the new one if force flag is specified.
+			if force {
+				err = os.Remove(filepath.Clean(localPath + "/kubeconfig"))
```
Comment (smira) [inline cmd/osctl/cmd/kubeconfig.go]: better to use `filepath.Join`

File: cmd/osctl/cmd/kubeconfig.go [hunk 2]
```diff
@@ -56,11 +60,19 @@ Kubeconfig will be written to PWD/kubeconfig or [local-path]/kubeconfig if speci
...
+				if err != nil && !os.IsNotExist(err) {
+					helpers.Fatalf("error deleting existing kubeconfig: %s", err)
+				}
+			}
 			extractTarGz(filepath.Clean(localPath), r)
```
Comment (smira) [inline cmd/osctl/cmd/kubeconfig.go]: I feel this whole thing better to be rewritten without `extractTarGz`, as this a bit of overkill for a single file

Comment (smira) [reply cmd/osctl/cmd/kubeconfig.go]: we have https://github.com/talos-systems/talos/blob/master/cmd/osctl/pkg/client/client.go#L166 wrapper method which can be used to simply write down `kubeconfig` where appropriate (and hopefully one day merge it into user's kubeconfig)

File: cmd/osctl/cmd/kubeconfig.go (hunk 1 above)
Comment (rsmitty) [reply cmd/osctl/cmd/kubeconfig.go]: fixed this up a bit to clean up the localpath and then use filepath.Join if force is supplied.

File: cmd/osctl/cmd/kubeconfig.go (hunk 2 above)
Comment (rsmitty) [reply cmd/osctl/cmd/kubeconfig.go]: Seems fair, but sounds like it's a bit larger overhaul of this kubeconfig command. I'll throw it in the backlog.

File: cmd/osctl/cmd/kubeconfig.go [hunk 3]
```diff
@@ -55,12 +57,21 @@ Kubeconfig will be written to PWD/kubeconfig or [local-path]/kubeconfig if speci
 			if len(args) == 1 {
 				localPath = args[0]
 			}
+			localPath = filepath.Clean(localPath)
 
-			extractTarGz(filepath.Clean(localPath), r)
+			// Drop the existing kubeconfig before writing the new one if force flag is specified.
+			if force {
+				err = os.Remove(filepath.Join(localPath, "/kubeconfig"))
```
Comment (andrewrynhard) [inline cmd/osctl/cmd/kubeconfig.go]: super nit: the extra `/` in `filepath.Join(localPath, "/kubeconfig")` isn't needed.

File: cmd/osctl/cmd/kubeconfig.go (hunk 2 above)
Comment (andrewrynhard) [reply cmd/osctl/cmd/kubeconfig.go]: Wonder if the new read api would be better here? It supports streaming. But agree something for a follow-up.

File: cmd/osctl/cmd/kubeconfig.go (hunk 3 above)
Comment (rsmitty) [reply cmd/osctl/cmd/kubeconfig.go]: fixed, hopefully ready for merge now.

**Pull request merged**